frontend/src/Components/App.tsx

```diff
@@ -0,0 +1,195 @@
+import React from "react";
+import {useState, useEffect} from "react";
+
+import { 
+  Card, 
+  CardActionArea, 
+  CardActions, 
+  CardContent, 
+  CardMedia, 
+  Button, 
+  Typography,
+  Container,
+  Paper,
+  CssBaseline,
+  Grid
+} from '@material-ui/core';
+
+import Error from "./Error";
+import Loading from "./Loading";
+import Product from "../Interfaces/Product";
+import PopupBox from "./PopupBox";
+
+import AddShoppingCartIcon from '@material-ui/icons/AddShoppingCartOutlined';
+import DeleteForeverIcon from '@material-ui/icons/DeleteForeverOutlined';
+import "../Styles/App.css"
+import APIService from '../APIService'
+import PopupBoxProps from "../Interfaces/PopupBoxProps";
+import Cookies from "../Cookies"
+
+export default function App(){
+
+	const [error, setError] = useState<string>("");
+	const [isLoaded, setIsLoaded] = useState<boolean>(false);
+	const [items, setItems] = useState<Product[]>([]);
+	const [cart, setCart] = useState<number[]>([]);
+	const [popupProps, setPopupProps] = useState<PopupBoxProps | undefined>();
+
+	function updateStorageCart(){
+		const storageCart = localStorage.getItem("cart")
+		if(storageCart !== null && storageCart !== ""){
+			setCart(JSON.parse(storageCart))
+		}
+	}
+
+	function updateStateItems(){
+		APIService.Products.list().then(
+			result => {
+				setIsLoaded(true)
+				setItems(result)
+			}
+		).catch(
+			e => {
+				setIsLoaded(true)
+				setError(e)
+			}
+		)
+	}
+
+	function closePopup() {
+		setPopupProps(undefined)
+	}
+
+	useEffect(() => {
+		updateStateItems()
+		updateStorageCart()
+	},[])
+
+	function addProductToCart(id: number) {
+		let arr = cart.length === 0 ? [] : [...cart]
```

Comment:
when an array is declared with `const`, you can still change the value of its elements. You just can't resign the name of it.

examples:
```
const arr = []
arr.push(1)
arr.push(2)
console.log(arr) // prints [1, 2]
```
```
const arr = []
arr = [1, 2] // throws error!
```